src/Microsoft.Dnx.Runtime/NuGet/Utility/DotNetGenerationMapping.cs

```diff
@@ -0,0 +1,82 @@
+            { new FrameworkName(VersionUtility.DnxFrameworkIdentifier, new Version(4, 5, 2)), new Version(5, 2) },
+            { new FrameworkName(VersionUtility.DnxFrameworkIdentifier, new Version(4, 6)), new Version(5, 3) },
+            { new FrameworkName(VersionUtility.DnxFrameworkIdentifier, new Version(4, 6, 1)), new Version(5, 4) },
+
+            // wpN
+            { new FrameworkName(VersionUtility.WindowsPhoneFrameworkIdentifier, new Version(8, 0)), new Version(5, 1) },
+            { new FrameworkName(VersionUtility.WindowsPhoneFrameworkIdentifier, new Version(8, 1)), new Version(5, 2) },
+
+            // wpaN
+            { new FrameworkName(VersionUtility.WindowsPhoneAppFrameworkIdentifier, new Version(8, 1)), new Version(5, 2) }
+        };
+
+        public static IEnumerable<FrameworkName> Expand(FrameworkName input)
+        {
+            // Try to convert the project framework into an equivalent target framework
+            // If the identifiers didn't match, we need to see if this framework has an equivalent framework that DOES match.
+            // If it does, we use that from here on.
+            // Example:
+            //  If the Project Targets ASP.Net, Version=5.0. It can accept Packages targetting .NETFramework, Version=4.5.1
+            //  so since the identifiers don't match, we need to "translate" the project target framework to .NETFramework
+            //  however, we still want direct ASP.Net == ASP.Net matches, so we do this ONLY if the identifiers don't already match
+
+            yield return input;
+
+            // dnxcoreN -> dotnetY
+            if (input.Identifier.Equals(VersionUtility.DnxCoreFrameworkIdentifier))
+            {
+                yield return GetGeneration(input);
+            }
+            // dnxN -> netN -> dotnetY
+            else if (input.Identifier.Equals(VersionUtility.DnxFrameworkIdentifier))
+            {
+                yield return new FrameworkName(VersionUtility.NetFrameworkIdentifier, input.Version);
+                yield return GetGeneration(input);
+            }
+            // netcore50 (universal windows apps) -> wpa81 -> dotnetY
+            else if (input.Identifier.Equals(VersionUtility.NetCoreFrameworkIdentifier) && input.Version.Major == 5 && input.Version.Minor == 0)
+            {
+                yield return new FrameworkName(VersionUtility.WindowsPhoneAppFrameworkIdentifier, new Version(8, 1));
+                yield return GetGeneration(input);
+            }
+            // netN (where N > 4.5) => dotnetY
+            else if(input.Identifier.Equals(VersionUtility.NetFrameworkIdentifier) && input.Version >= _version45)
+            {
+                yield return GetGeneration(input);
+            }
+        }
+
+        public static FrameworkName GetGeneration(FrameworkName input)
+        {
+            var version = _mappings[input];
```

Comment:
I originally had more defensive code but a) it makes the generator above complicated and b) we only ever call GetGeneration with frameworks we **know** should have mappings. I'll make it private (should always have been so) and give defensive coding one more shot though, I have an idea that might work